### PR TITLE
Add NodeReuse property to DotNetMSBuildSettings

### LINKED ISSUE #4144: `DotNetMSBuildSettings` is missing `NodeReuse`
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched [issues](https://github.com/cake-build/cake/issues) to ensure it has not already been reported

### Cake runner

Cake .NET Tool

### Cake version

3.0.0

### Operating system

N/A

### Operating system architecture

N/A

### CI Server

_No response_

### What are you seeing?

`DotNetMSBuildSettings` is missing the `NodeReuse` property that `MSBuildSettings` has, necessitating the use of a `Directory.Build.rsp` file to set this flag.

### What is expected?

`DotNetMSBuildSettings` should have a `NodeReuse` property.

### Steps to Reproduce

N/A

### Output log

_No response_

### PR DESCRIPTION
**Pull request by Marusyk**
Add NodeReuse property to DotNetMSBuildSettings and tests

Fixes #4144 

Please review,
Thank you in advance

### CODE REVIEW

File: src/Cake.Common/Tools/DotNet/MSBuild/MSBuildArgumentBuilderExtensions.cs [hunk 1]
```diff
@@ -203,6 +203,12 @@ public static void AppendMSBuildSettings(this ProcessArgumentBuilder builder, Do
                 msBuilder.AppendMSBuildSwitch("property", $"ContinuousIntegrationBuild={continuousIntegrationBuild}");
             }
 
+            // Re-use of MSBuild nodes?
+            if (settings.NodeReuse != null)
+            {
+                msBuilder.Append(string.Concat("/nr:", settings.NodeReuse.Value ? "true" : "false"));
+            }
```
Comment (augustoproiete) [inline src/Cake.Common/Tools/DotNet/MSBuild/MSBuildArgumentBuilderExtensions.cs:210]:
```suggestion
            // Re-use of MSBuild nodes?
            if (settings.NodeReuse.HasValue)
            {
                msBuilder.Append(string.Concat("/nodeReuse:", settings.NodeReuse.Value ? "true" : "false"));
            }
```

Let's use the longer name version to make it easier to understand

File: src/Cake.Common.Tests/Unit/Tools/DotNet/MSBuild/DotNetMSBuildBuilderTests.cs [hunk 2]
```diff
@@ -801,6 +802,20 @@ public void Should_Add_Host_Arguments()
...
+            {
+                // Given
+                var fixture = new DotNetMSBuildBuilderFixture();
+                fixture.Settings.NodeReuse = true;
+
+                // When
+                var result = fixture.Run();
+
+                // Then
+                Assert.Equal("msbuild /nr:true", result.Args);
```
Comment (augustoproiete) [inline src/Cake.Common.Tests/Unit/Tools/DotNet/MSBuild/DotNetMSBuildBuilderTests.cs:817]:
```suggestion
                Assert.Equal("msbuild /nodeReuse:true", result.Args);
```

Let's use the longer name version to make it easier to understand

**Review by augustoproiete — CHANGES_REQUESTED** [review]
LGTM overall. Small change on the command-line

**Review by augustoproiete — APPROVED** [review]
LGTM

**Comment by alexrp** [comment]
@augustoproiete seems like this one somehow got stuck pending approval despite having approval? 🤔

**Review by devlead — APPROVED** [review]
LGTM 👍

**Comment by devlead** [comment]
@Marusyk your changes have been merged, thanks for your contribution 👍

**Pull request merged**